src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,192 @@
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void passwordWithSpace_NotOk() {
+        user.setPassword(RegistrationServiceTestConstants.PASSWORD_WITH_SPACE);
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void minPassword_Ok() {
+        user.setPassword(RegistrationServiceTestConstants.MIN_PASSWORD);
+        User actual = registrationService.register(user);
+        Assertions.assertEquals(user, actual);
+    }
+
+    @Test
+    void overMinPassword_Ok() {
+        user.setPassword(RegistrationServiceTestConstants.OVER_MIN_PASSWORD);
+        User actual = registrationService.register(user);
+        Assertions.assertEquals(user, actual);
+    }
+
+    @Test
+    void nullAge_NotOk() {
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            user.setAge(null);
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void notEnoughAge_NotOk() {
+        user.setAge(RegistrationServiceTestConstants.NOT_ENOUGH_AGE);
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void negativeAge_NotOk() {
+        user.setAge(RegistrationServiceTestConstants.NEGATIVE_AGE);
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void minAge_Ok() {
+        user.setAge(RegistrationServiceTestConstants.MIN_AGE);
+        User actual = registrationService.register(user);
+        Assertions.assertEquals(user, actual);
+    }
+
+    @Test
+    void overMinAge_Ok() {
+        user.setAge(RegistrationServiceTestConstants.OVER_MIN_AGE);
+        User actual = registrationService.register(user);
+        Assertions.assertEquals(user, actual);
+    }
+
+    private static class RegistrationServiceTestConstants {
+        private static final String OVER_MIN_LOGIN = "roberto";
+        private static final String SHORT_LOGIN = "bob";
+        private static final String MIN_LOGIN = "morris";
+        private static final String LOGIN_WITH_SPACE = "gary  ";
+        private static final String EMPTY_LOGIN = "";
+        private static final int OVER_MIN_AGE = 22;
+        private static final int MIN_AGE = 18;
+        private static final int NOT_ENOUGH_AGE = 12;
+        private static final int NEGATIVE_AGE = -10;
+        private static final String OVER_MIN_PASSWORD = "robinson134";
+        private static final String SHORT_PASSWORD = "984";
+        private static final String MIN_PASSWORD = "123456";
+        private static final String PASSWORD_WITH_SPACE = "robi   ";
+        private static final String EMPTY_PASSWORD = "";
+    }
```

Comment:
I think @rovein meant that this class should be a regular one, not an inner class